deploy/examples/sqlitevfs-client.yaml

```diff
@@ -0,0 +1,178 @@
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: sqlitevfs
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: sqlitevfs
+      component: app
+  template:
+    metadata:
+      labels:
+        app: sqlitevfs
+        component: app
+    spec:
+      serviceAccountName: sqlitevfs-setup
+
+      initContainers:
+        ## Setup Ceph SQLite VFS
+        - name: setup
+          image: bitnami/kubectl:1.21.11
+          command:
+            - /bin/bash
+            - -c
+            - |
+              # Wait until sqlitevfs cephclient is ready
+              SECRET_NAME=`kubectl get cephclient ${CEPH_CLIENT_NAME} -o jsonpath='{.status.info.secretName}' -n ${CEPH_NS}`
+              while [ -z "${SECRET_NAME}" ]; do
+                sleep 3;
+                echo -e "[info] checking for cephclient [${CEPH_CLIENT_NAME}]...";
+                SECRET_NAME=`kubectl get cephclient ${CEPH_CLIENT_NAME} -o jsonpath='{.status.info.secretName}' -n ${CEPH_NS}`;
+              done;
+
+              echo -e "[info] found ceph client secret [${SECRET_NAME}] in namespace [${CEPH_NS}], extracting content...";
+              SECRET_CONTENT=`kubectl get secret ${SECRET_NAME} -o jsonpath="{.data.${CEPH_CLIENT_NAME}}" -n ${CEPH_NS} | base64 -d`;
+
+              echo -e "[info] writing ceph keyring...";
+              echo -e "[client.${CEPH_CLIENT_NAME}]" > /libsqliteceph/ceph.keyring;
+              echo -e "  key = ${SECRET_CONTENT}" >> /libsqliteceph/ceph.keyring;
+              echo -e "  caps mon = 'allow r'" >> /libsqliteceph/ceph.keyring;
+              echo -e "  caps osd = 'profile rbd pool=raid1, profile rbd pool=rook-raid-1-3n-host'" >> /libsqliteceph/ceph.keyring;
+
+              echo -e "retrieving mon endpoints...";
+              MON_ENDPOINTS=`kubectl get configmap rook-ceph-mon-endpoints -o jsonpath='{.data.data}' -n ${CEPH_NS} | sed 's/.=//g'`;
+
+              echo -e "[info] writing out ceph.conf...";
+              echo -e "[global]" > /libsqliteceph/ceph.conf;
+              echo -e "mon_host=${MON_ENDPOINTS}" >> /libsqliteceph/ceph.conf;
+
+              echo -e "[info] writing ceph args...";
+              echo -e "--id ${CEPH_CLIENT_NAME}" > /libsqliteceph/ceph.args
+
+              echo -e "[info] successfully set up temporary libsqliteceph credentials!";
+          env:
+            - name: CEPH_CLIENT_NAME
+              value: sqlitevfs
+            - name: CEPH_NS
+              value: rook-ceph
+          volumeMounts:
+            - name: libsqliteceph-config
+              mountPath: /libsqliteceph
+
+      containers:
+        - name: sqlitevfs
+          image: alpine:3.16.2
+          command:
+            - /bin/ash
+            - -c
+            - |
+              apk add ceph vim sqlite;
+
+              export CEPH_CONF=/libsqliteceph/ceph.conf;
+              export CEPH_KEYRING=/libsqliteceph/ceph.keyring;
+              export CEPH_ARGS=`cat /libsqliteceph/ceph.args`;
```

Comment:
no more args file?

Reply:
Oh sorry this got left in there!

Thank you for the thorough review :bow:

Reply:
- [ ] remove args file use